spec/controls.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { createDOM, mockMapFactory } from "./helpers";
+import { 
+    ONSZoomInControl, 
+    ONSZoomOutControl,
+    ONSResetControl, 
+    ONSFullScreenControl,
+} from "../src/ts/controls"
+import { MapState } from "../src/ts/context";
+
+describe("controls", () => {
+    describe("#ONSZoomInControl()", () => {
+        test("#onClick() should alter the DOM", () => {
+            document.body.innerHTML = createDOM(document);
+            const map = mockMapFactory();
+            const onsZoomInControl = new ONSZoomInControl(map, ".ons-ctrl-zoom-in");
+            onsZoomInControl.setup();
+            const button = document.querySelector(".ons-ctrl-zoom-in");
+            button?.dispatchEvent(new MouseEvent("click", {shiftKey: true}));
+        });
+    });
+    describe("#ONSZoomOutControl()", () => {
+        test("#onClick()", () => {
+            document.body.innerHTML = createDOM(document);
+            const map = mockMapFactory();
+            const onsZoomOutControl = new ONSZoomOutControl(map, ".ons-ctrl-zoom-out");
+            onsZoomOutControl.setup();
+            onsZoomOutControl.onClick({} as Event);
+            const button = document.querySelector(".ons-ctrl-zoom-out");
+            button?.dispatchEvent(new MouseEvent("click", {shiftKey: true}));
+        });
+    });
+    describe("#ONSResetControl()", () => {
+
+        test("#onClick()", () => {
+            document.body.innerHTML = createDOM(document);
+            const map = mockMapFactory();
+            const onsResetControl = new ONSResetControl(map, ".ons-ctrl-reset", "mapboxgl-ctrl-icon--disabled", [1,1], 6);
+            onsResetControl.setup();
+            // onsResetControl.onClick({} as Event);
+            const button = document.querySelector(".ons-ctrl-zoom-in");
+            button?.dispatchEvent(new MouseEvent("click", {shiftKey: true}));
+            const result = onsResetControl.iconElement.className;
+            expect(result).toEqual("mapboxgl-ctrl-icon mapboxgl-ctrl-icon--disabled")
+        });
+
+    });
+    describe("#ONSFullScreenControl()", () => {
+        xtest("#onClick()", () => {
```

Comment:
will you un comment that later on? this test is skipped by purpose?

Reply:
Yes, I will get this this before the official first release, thank you